src/vendors/saga.ts

```diff
@@ -0,0 +1,20 @@
+import { call, put, spawn, take } from 'redux-saga/effects'
+import { getVendors } from 'src/api/vendors'
+import { formatVendors } from 'src/utils/formatVendors'
+import { Actions, setLoading, setVendors } from 'src/vendors/actions'
+
+export function* watchFetchVendors(): any {
+  while (true) {
```

Comment:
How does this function not spam the server? I also noticed there is no error handling in this function. Can you make use of the notification saga that is built into valora that can show a message to the user that the fetch failed?

Reply:
Im not sure, this function is the same for when the app first loads to block the interface by showing a loading screen and loading the vendor data from the API then showing it. refreshing has the same effect that's why they both run this function.